CompleteDynamicPropertiesRector adds property, but parent is missing

It should add parent as well

```bash
[ERROR] Could not process "src/Service/MailService.php" file, due to:                                                  
         "PropertyProperty should always have Property as parent".
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -135,6 +135,7 @@ jobs:
 
     allow_failures:
         - name: ECS
+        - name: Rector
         - name: Standalone Run
         - name: Documentation
         - os: windows

--- a/packages/CodeQuality/src/Rector/Class_/CompleteDynamicPropertiesRector.php
+++ b/packages/CodeQuality/src/Rector/Class_/CompleteDynamicPropertiesRector.php
@@ -207,9 +207,7 @@ PHP
                 continue;
             }
 
-            $propertyBuilder = $this->builderFactory->property($propertyName);
-            $propertyBuilder->makePublic();
-            $property = $propertyBuilder->getNode();
+            $property = $this->nodeFactory->createPublicProperty($propertyName);
 
             if ($this->isAtLeastPhpVersion(PhpVersionFeature::TYPED_PROPERTIES)) {
                 $phpStanNode = $this->staticTypeMapper->mapPHPStanTypeToPhpParserNode($propertyType);

--- a/packages/Doctrine/src/NodeFactory/EntityUuidNodeFactory.php
+++ b/packages/Doctrine/src/NodeFactory/EntityUuidNodeFactory.php
@@ -5,7 +5,6 @@ declare(strict_types=1);
 namespace Rector\Doctrine\NodeFactory;
 
 use Nette\Utils\Strings;
-use PhpParser\BuilderFactory;
 use PhpParser\Comment\Doc;
 use PhpParser\Node;
 use PhpParser\Node\Expr\Assign;
@@ -18,14 +17,10 @@ use PhpParser\Node\Stmt\Property;
 use Ramsey\Uuid\Uuid;
 use Rector\Doctrine\PhpDocParser\Ast\PhpDoc\PhpDocTagNodeFactory;
 use Rector\NodeTypeResolver\PhpDoc\NodeAnalyzer\DocBlockManipulator;
+use Rector\PhpParser\Node\NodeFactory;
 
 final class EntityUuidNodeFactory
 {
-    /**
-     * @var BuilderFactory
-     */
-    private $builderFactory;
-
     /**
      * @var PhpDocTagNodeFactory
      */
@@ -36,22 +31,24 @@ final class EntityUuidNodeFactory
      */
     private $docBlockManipulator;
 
+    /**
+     * @var NodeFactory
+     */
+    private $nodeFactory;
+
     public function __construct(
-        BuilderFactory $builderFactory,
         PhpDocTagNodeFactory $phpDocTagNodeFactory,
-        DocBlockManipulator $docBlockManipulator
+        DocBlockManipulator $docBlockManipulator,
+        NodeFactory $nodeFactory
     ) {
-        $this->builderFactory = $builderFactory;
         $this->phpDocTagNodeFactory = $phpDocTagNodeFactory;
         $this->docBlockManipulator = $docBlockManipulator;
+        $this->nodeFactory = $nodeFactory;
     }
 
     public function createTemporaryUuidProperty(): Property
     {
-        $uuidPropertyBuilder = $this->builderFactory->property('uuid');
-        $uuidPropertyBuilder->makePrivate();
-
-        $uuidProperty = $uuidPropertyBuilder->getNode();
+        $uuidProperty = $this->nodeFactory->createPrivateProperty('uuid');
 
         $this->decoratePropertyWithUuidAnnotations($uuidProperty, true, false);
 

--- a/packages/RemovingStatic/src/Rector/Class_/StaticTypeToSetterInjectionRector.php
+++ b/packages/RemovingStatic/src/Rector/Class_/StaticTypeToSetterInjectionRector.php
@@ -70,17 +70,17 @@ final class CheckoutEntityFactory
      * @var SomeStaticClass
      */
     private $someStaticClass;
-    
+
     public function setSomeStaticClass(SomeStaticClass $someStaticClass)
     {
         $this->someStaticClass = $someStaticClass;
     }
-    
+
     public function run()
     {
         return $this->someStaticClass->go();
     }
-}               
+}
 PHP
                 ,
                 [
@@ -153,10 +153,7 @@ PHP
 
             $setEntityFactoryMethod = $this->createSetEntityFactoryClassMethod($variableName, $param, $assign);
 
-            $entityFactoryPropertyBuilder = $this->builderFactory->property($variableName);
-            $entityFactoryPropertyBuilder->makePrivate();
-            $entityFactoryProperty = $entityFactoryPropertyBuilder->getNode();
-
+            $entityFactoryProperty = $this->nodeFactory->createPrivateProperty($variableName);
             $this->docBlockManipulator->changeVarTag($entityFactoryProperty, $objectType);
 
             $class->stmts = array_merge([$entityFactoryProperty, $setEntityFactoryMethod], $class->stmts);

--- a/packages/RemovingStatic/src/UniqueObjectFactoryFactory.php
+++ b/packages/RemovingStatic/src/UniqueObjectFactoryFactory.php
@@ -22,6 +22,7 @@ use Rector\Exception\ShouldNotHappenException;
 use Rector\Naming\PropertyNaming;
 use Rector\NodeTypeResolver\PhpDoc\NodeAnalyzer\DocBlockManipulator;
 use Rector\NodeTypeResolver\StaticTypeMapper;
+use Rector\PhpParser\Node\NodeFactory;
 use Rector\PhpParser\Node\Resolver\NameResolver;
 
 final class UniqueObjectFactoryFactory
@@ -51,18 +52,25 @@ final class UniqueObjectFactoryFactory
      */
     private $staticTypeMapper;
 
+    /**
+     * @var NodeFactory
+     */
+    private $nodeFactory;
+
     public function __construct(
         NameResolver $nameResolver,
         BuilderFactory $builderFactory,
         PropertyNaming $propertyNaming,
         DocBlockManipulator $docBlockManipulator,
-        StaticTypeMapper $staticTypeMapper
+        StaticTypeMapper $staticTypeMapper,
+        NodeFactory $nodeFactory
     ) {
         $this->nameResolver = $nameResolver;
         $this->builderFactory = $builderFactory;
         $this->propertyNaming = $propertyNaming;
         $this->docBlockManipulator = $docBlockManipulator;
         $this->staticTypeMapper = $staticTypeMapper;
+        $this->nodeFactory = $nodeFactory;
     }
 
     public function createFactoryClass(Class_ $class, ObjectType $objectType): Class_
@@ -110,11 +118,8 @@ final class UniqueObjectFactoryFactory
         $properties = [];
 
         $propertyName = $this->propertyNaming->fqnToVariableName($objectType);
-        $propertyBuilder = $this->builderFactory->property($propertyName);
-        $propertyBuilder->makePrivate();
 
-        $property = $propertyBuilder->getNode();
-
+        $property = $this->nodeFactory->createPrivateProperty($propertyName);
         $this->docBlockManipulator->changeVarTag($property, $objectType);
 
         $properties[] = $property;

--- a/packages/Symfony/src/Rector/Class_/MakeCommandLazyRector.php
+++ b/packages/Symfony/src/Rector/Class_/MakeCommandLazyRector.php
@@ -10,7 +10,6 @@ use PhpParser\Node\Expr\MethodCall;
 use PhpParser\Node\Expr\StaticCall;
 use PhpParser\Node\Stmt\Class_;
 use PhpParser\Node\Stmt\Expression;
-use PhpParser\Node\Stmt\Property;
 use PHPStan\Type\StringType;
 use Rector\Rector\AbstractRector;
 use Rector\RectorDefinition\CodeSample;
@@ -77,7 +76,10 @@ PHP
             return null;
         }
 
-        $defaultNameProperty = $this->createDefaultNameProperty($commandName);
+        $defaultNameProperty = $this->nodeFactory->createStaticProtectedPropertyWithDefault(
+            'defaultName',
+            $commandName
+        );
 
         $node->stmts = array_merge([$defaultNameProperty], (array) $node->stmts);
 
@@ -124,16 +126,6 @@ PHP
         $this->removeConstructorIfHasOnlySetNameMethodCall($class);
 
         return $commandName;
-    }
-
-    private function createDefaultNameProperty(Node $commandNameNode): Property
-    {
-        $propertyBuilder = $this->builderFactory->property('defaultName');
-        $propertyBuilder->makeProtected();
-        $propertyBuilder->makeStatic();
-        $propertyBuilder->setDefault($commandNameNode);
-
-        return $propertyBuilder->getNode();
     }
 
     private function matchCommandNameNodeInConstruct(Expr $expr): ?Node

--- a/src/PhpParser/Node/NodeFactory.php
+++ b/src/PhpParser/Node/NodeFactory.php
@@ -7,6 +7,7 @@ namespace Rector\PhpParser\Node;
 use PhpParser\BuilderFactory;
 use PhpParser\BuilderHelpers;
 use PhpParser\Comment\Doc;
+use PhpParser\Node;
 use PhpParser\Node\Arg;
 use PhpParser\Node\Expr;
 use PhpParser\Node\Expr\Array_;
@@ -167,7 +168,11 @@ final class NodeFactory
             $propertyBuilder->setDocComment($docComment);
         }
 
-        return $propertyBuilder->getNode();
+        $property = $propertyBuilder->getNode();
+
+        $this->decorateParentPropertyProperty($property);
+
+        return $property;
     }
 
     /**
@@ -221,6 +226,42 @@ final class NodeFactory
             new Identifier('__construct'),
             $this->convertParamNodesToArgNodes($params)
         );
+    }
+
+    public function createPrivateProperty(string $name): Property
+    {
+        $propertyBuilder = $this->builderFactory->property($name);
+        $propertyBuilder->makePrivate();
+        $property = $propertyBuilder->getNode();
+
+        $this->decorateParentPropertyProperty($property);
+
+        return $property;
+    }
+
+    public function createStaticProtectedPropertyWithDefault(string $name, Node $node): Property
+    {
+        $propertyBuilder = $this->builderFactory->property($name);
+        $propertyBuilder->makeProtected();
+        $propertyBuilder->makeStatic();
+        $propertyBuilder->setDefault($node);
+
+        $property = $propertyBuilder->getNode();
+
+        $this->decorateParentPropertyProperty($property);
+
+        return $property;
+    }
+
+    public function createPublicProperty(string $name): Property
+    {
+        $propertyBuilder = $this->builderFactory->property($name);
+        $propertyBuilder->makePublic();
+        $property = $propertyBuilder->getNode();
+
+        $this->decorateParentPropertyProperty($property);
+
+        return $property;
     }
 
     /**
@@ -284,5 +325,12 @@ final class NodeFactory
         }
 
         return $args;
+    }
+
+    private function decorateParentPropertyProperty(Property $property): void
+    {
+        // complete property property parent, needed for other operations
+        $propertyProperty = $property->props[0];
+        $propertyProperty->setAttribute(AttributeKey::PARENT_NODE, $property);
     }
 }

